src/porepy/models/solution_strategy.py

```diff
@@ -173,6 +186,14 @@ def initial_condition(self) -> None:
         vals = np.zeros(self.equation_system.num_dofs())
         self.equation_system.set_variable_values(vals, to_iterate=True, to_state=True)
 
+        # Overwrite the initial condition in case of a restart
```

Comment:
If a child method first calls this method through `super` and then does some modifications (see for instance contact forces in contact mechanics), one would have to overwrite `initial_condition` in the child method? I think this should be better documented.

Reply:
After finishing answerign to your questions, I want to put this question on hold. I will have to debug some attempt to disconnect the restart from initial condition. Because I absolutely see that one has to be very careful what one is doing if not applying conditionals as time_manager.is_init() as in test_poromechanics. However, the switch from the combined use of initial condition and restart (despite the use of is_init) results in a different simulation result and I have aboslutely no clue why. So I will have to investigate and I will report further later. My feeling is though a cleaner separation for more robustness (instead of a comment which not every user will know of) will be the better solution.

Reply:
With the most recent design for how to restart a simulation, this discussion becomes obsolete. I therefore resolve this conversation. And open a new one below.